test/e2e/performanceprofile/functests/14_arm/kernelpagesize.go

```diff
@@ -0,0 +1,139 @@
+package __arm
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/ptr"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	performancev2 "github.com/openshift/cluster-node-tuning-operator/pkg/apis/performanceprofile/v2"
+	testutils "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/infrastructure"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/label"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/nodes"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/poolname"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/profiles"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/profilesupdate"
+)
+
+const (
+	kernelPageSize4k       = "4k"
+	kernelPageSize64k      = "64k"
+	kernelPageSizeBytes4k  = "4096"
+	kernelPageSizeBytes64k = "65536"
+)
+
+var _ = Describe("[rfe_id:80342] kernelPageSize configuration validation on aarch64", Label(string(label.OpenShift)), Ordered, func() {
+	var (
+		workerRTNodes               []corev1.Node
+		perfProfile, initialProfile *performancev2.PerformanceProfile
+		poolName                    string
+		err                         error
+		ctx                         context.Context = context.Background()
+		isArm                       bool
+		workerRTNode                corev1.Node
+	)
+
+	BeforeAll(func() {
+		workerRTNodes, err = nodes.GetByLabels(testutils.NodeSelectorLabels)
+		Expect(err).ToNot(HaveOccurred())
+
+		workerRTNodes, err = nodes.MatchingOptionalSelector(workerRTNodes)
+		Expect(err).ToNot(HaveOccurred())
+		workerRTNode = workerRTNodes[0]
+
+		perfProfile, err = profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+		initialProfile = perfProfile.DeepCopy()
+
+		poolName = poolname.GetByProfile(ctx, perfProfile)
+		Expect(err).ToNot(HaveOccurred())
+
+		isArm, err = infrastructure.IsARM(ctx, &workerRTNode)
+		Expect(err).ToNot(HaveOccurred())
+		if !isArm {
+			Skip("This test requires an aarch64 arm CPU")
+		}
+
+		By("Make sure that the performance profile starts the test suite with 4k kernelPageSize and real time kernel disabled")
+		currentKernelPageSize, err := getKernelPageSizeFromNode(ctx, workerRTNode)
+		Expect(err).ToNot(HaveOccurred())
+
+		if *perfProfile.Spec.RealTimeKernel.Enabled || currentKernelPageSize != kernelPageSize4k {
```

Comment:
This is a pointer dereference here, so I would check that it's not `nil` first

Reply:
Thanks Talor,  Agree with Makefile targets . Currently tests need to be run manually